app/models/mixins/miq_request_mixin.rb

```diff
@@ -169,4 +169,13 @@ def request_dialog(action_name)
     dialog = ResourceActionWorkflow.new(values, get_user, ra, {}).dialog
     DialogSerializer.new.serialize(Array[dialog]).first
   end
+
+  def set_option(key, value)
+    current_class = options[key.to_sym].class
+    if value.class != current_class && current_class != NilClass
+      raise ArgumentError, "Incorrect type key: #{key}, value: #{value}, expected type: #{current_class}"
+    end
+    options[key.to_sym] = value
+    update_attribute(:options, options)
```

Comment:
Use `update(:options => options)`

This might be crazy, but `tap` would make this a 1-liner:
`update(:options => options.tap { |o| o[key.to_sym] = value })`